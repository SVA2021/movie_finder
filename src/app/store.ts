import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
