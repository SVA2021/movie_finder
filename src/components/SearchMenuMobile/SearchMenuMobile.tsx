import {ExpandMore, Search} from '@mui/icons-material';
import {Accordion, AccordionDetails, AccordionSummary} from '@mui/material';
import {FC} from 'react';
import {SearchField, StyledTextItem} from '../../components';

export const SearchMenuMobile: FC = () => {
  return (
    <Accordion data-testid={'search-menu-mobile'}>
      <AccordionSummary
        expandIcon={<ExpandMore />}
        sx={{p: 0}}
        id="panel-search-menu-btn"
        aria-controls="panel-search-menu-btn"
        data-testid={'search-menu-mobile-btn'}
      >
        <StyledTextItem ><Search fontSize='medium' />Поиск</StyledTextItem>
      </AccordionSummary>
      <AccordionDetails
        sx={{p: 0}}
        id="panel-search-menu-content"
        data-testid={'search-menu-mobile-content'}
      >
        <SearchField />
      </AccordionDetails>
    </Accordion>
  );
}