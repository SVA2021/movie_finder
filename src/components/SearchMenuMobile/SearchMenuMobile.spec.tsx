import {fireEvent, render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from "react-router-dom";
import {store} from "../../app/store";
import {SearchMenuMobile} from "./SearchMenuMobile";

describe('render SearchMenuMobile', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <SearchMenuMobile />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('search-menu-mobile')).toMatchSnapshot();
  });

  it('check button open menu', () => {
    setUp();
    fireEvent.click(screen.getByTestId('search-menu-mobile-btn'))
    expect(screen.getByTestId('search-menu-mobile-content')).toBeInTheDocument();
  });

})