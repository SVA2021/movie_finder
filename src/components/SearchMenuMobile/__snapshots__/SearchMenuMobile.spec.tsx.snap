// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchMenuMobile snapshot test 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  data-testid="search-menu-mobile"
>
  <div
    aria-controls="panel-search-menu-btn"
    aria-expanded="false"
    class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-9ci07q-MuiButtonBase-root-MuiAccordionSummary-root"
    data-testid="search-menu-mobile-btn"
    id="panel-search-menu-btn"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        class="MuiTypography-root MuiTypography-h3 css-14at1vf-MuiTypography-root"
        data-testid="styled-text-item"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        Поиск
      </p>
    </div>
    <div
      class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel-search-menu-btn"
          class="MuiAccordion-region"
          id="panel-search-menu-btn"
          role="region"
        >
          <div
            class="MuiAccordionDetails-root css-1jv654d-MuiAccordionDetails-root"
            data-testid="search-menu-mobile-content"
            id="panel-search-menu-content"
          >
            <form
              class="MuiBox-root css-hpgf8j"
            >
              <div
                class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedEnd css-1qmitm0-MuiInputBase-root-MuiInput-root"
              >
                <input
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  placeholder="Movies, series, keywords..."
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wgghly-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
